plugin_tests/assetstore_test.py

```diff
@@ -0,0 +1,633 @@
+                            params=self.dbParams)
+        self.assertStatusOk(resp)
+        assetstore1 = resp.json
+        # Import some tables from postgres
+        params = {
+            'parentId': str(self.publicFolder['_id']),
+            'parentType': 'folder',
+            'table': json.dumps([
+                'towns', {'table': 'edges', 'schema': 'tiger'}]),
+        }
+        resp = self.request(
+            path='/database_assetstore/%s/import' % str(assetstore1['_id']),
+            method='PUT', user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        # Mark the towns database as not imported
+        townItem = list(self.model('item').textSearch('towns', user=self.admin,
+                                                      limit=1))[0]
+        townFile = list(self.model('item').childFiles(item=townItem))[0]
+        del townFile['databaseMetadata']['imported']
+        self.model('file').save(townFile)
+        # We shouldn't be allowed to delete towns
+        with six.assertRaisesRegex(self, Exception,
+                                   'Database assetstores are read only'):
+            resp = self.request(path='/item/%s' % str(townItem['_id']),
+                                method='DELETE', user=self.admin)
+        # If we remark it as imported, we can
+        townFile['databaseMetadata']['imported'] = True
+        self.model('file').save(townFile)
+        resp = self.request(path='/item/%s' % str(townItem['_id']),
+                            method='DELETE', user=self.admin)
+        self.assertStatusOk(resp)
+        # We can't upload
+        adapter = assetstore_utilities.getAssetstoreAdapter(assetstore1)
+        with six.assertRaisesRegex(self, Exception,
+                                   'Database assetstores are read only'):
+            adapter.initUpload({})
+        with six.assertRaisesRegex(self, Exception,
+                                   'Database assetstores are read only'):
+            adapter.finalizeUpload({}, {})
+
+    def testAssetstoreDownload(self):
+        # Create assetstore
+        resp = self.request(path='/assetstore', method='POST', user=self.admin,
+                            params=self.dbParams)
+        self.assertStatusOk(resp)
+        assetstore1 = resp.json
+        # Import some tables from postgres
+        params = {
+            'parentId': str(self.publicFolder['_id']),
+            'parentType': 'folder',
+            'table': 'towns',
+            'format': 'list',
+            'fields': 'town,pop2010',
+            'limit': '10'
+        }
+        resp = self.request(
+            path='/database_assetstore/%s/import' % str(assetstore1['_id']),
+            method='PUT', user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        townItem = list(self.model('item').textSearch('towns', user=self.admin,
+                                                      limit=1))[0]
```

Comment:
I think most girder devs have expressed a preference against visual indentation of params due to readability concerns.